protocol/json-schemas/tbdex-message-schema.json

```diff
@@ -0,0 +1,276 @@
+                "kind"
+            ]
+        },
+        "paymentInstructions": {
+            "type": "object",
+            "properties": {
+                "payin": {
+                    "$ref": "#/definitions/paymentInstruction"
+                },
+                "payout": {
+                    "$ref": "#/definitions/paymentInstruction"
+                }
+            }
+        },
+        "paymentInstruction": {
+            "type": "object",
+            "properties": {
+                "link": {
+                    "type": "string"
+                },
+                "instruction": {
+                    "type": "string"
+                }
+            }
+        }
+    },
+    "allOf": [
+        {
+            "$ref": "#/definitions/metadata"
+        },
+        {
+            "type": "object",
+            "properties": {
+                "type": {
+                    "enum": [
+                        "offering",
+                        "rfq",
+                        "quote",
+                        "order",
+                        "orderStatus"
+                    ]
+                }
+            }
+        },
+        {
+            "oneOf": [
+                {
+                    "type": "object",
+                    "required": [
+                        "body"
+                    ],
+                    "properties": {
+                        "body": {
+                            "$ref": "#/definitions/offering"
+                        }
+                    }
+                },
+                {
+                    "type": "object",
+                    "required": [
+                        "body"
+                    ],
+                    "properties": {
+                        "body": {
+                            "$ref": "#/definitions/rfq"
+                        }
+                    }
+                },
+                {
+                    "type": "object",
+                    "required": [
+                        "body"
+                    ],
+                    "properties": {
+                        "body": {
+                            "$ref": "#/definitions/quote"
+                        }
+                    }
+                },
+                {
+                    "type": "object",
+                    "required": [
+                        "body"
+                    ],
+                    "properties": {
+                        "body": {
+                            "$ref": "#/definitions/order"
+                        }
+                    }
+                },
+                {
+                    "type": "object",
+                    "required": [
+                        "body"
+                    ],
+                    "properties": {
+                        "body": {
+                            "$ref": "#/definitions/orderStatus"
+                        }
+                    }
+                }
+            ]
+        }
+    ]
+}
```

Comment:
here's what chad said about `$schema`: 
<img width="844" alt="image" src="https://github.com/TBD54566975/tbdex-protocol/assets/118510103/5bf681a8-03af-4da6-b330-7d046b8acbd8">


i think including $schema would be good. i'm not really sure what we'd use for $id right now. maybe if this schema is published somewhere later, we can use the url for $id later?

Reply:
@jiyoontbd or @mistermoe the examples of $schemas i've seen are all URLs - is there one which makes sense for us?